### PR TITLE
add a debug flag for command line to print http headers

### PR DESCRIPTION
**Pull request by sondhayni-rfqa**

### CODE REVIEW

File: rainforest-cli_test.go [hunk 1]
```diff
@@ -38,6 +45,63 @@ func TestShuffleFlags(t *testing.T) {
 	}
 }
 
+func TestDebugFlag(t *testing.T) {
+
```
Comment (epaulet) [inline rainforest-cli_test.go]: Unnecessary newline in the beginning of the function here as well

File: rainforest-cli_test.go [hunk 2]
```diff
@@ -38,6 +45,63 @@ func TestShuffleFlags(t *testing.T) {
...
+			tag:    "star",
+			token:  "testToken123",
+			method: "POST",
+		},
+	}
+
+	for _, testCase := range testCases {
+		c := newFakeContext(testCase.mappings, testCase.args)
+		client := rainforest.NewClient(testCase.token, c.Bool("debug"))
+		client.BaseURL, _ = url.Parse("https://example.org")
```
Comment (epaulet) [inline rainforest-cli_test.go]: Test should fail if an error is returned here.

File: rainforest-cli_test.go [hunk 3]
```diff
@@ -38,6 +45,63 @@ func TestShuffleFlags(t *testing.T) {
 	}
 }
 
+func TestDebugFlag(t *testing.T) {
```
Comment (epaulet) [inline rainforest-cli_test.go]: This test should be in the rainforest package.

Comment (sondhayni-rfqa) [inline rainforest-cli_test.go]: I'm not sure how I would create the fakeContext/access related functions in that case?

File: rainforest/rainforest_test.go [hunk 4]
```diff
@@ -137,3 +139,74 @@ func TestDo(t *testing.T) {
...
+			debug:  false,
+			tag:    "star",
+			token:  "testToken123",
+			method: "POST",
+		},
+	}
+
+	for _, testCase := range testCases {
+		c := fakeContext{mappings: testCase.mappings, args: testCase.args}
+		client := NewClient(testCase.token, c.Bool("debug"))
```
Comment (epaulet) [inline rainforest/rainforest_test.go]: You don't need to create a CLI context here. Just pass in true or false.

**Comment by sondhayni-rfqa** [comment]
Sample output:
```
$ ./rainforest-cli --token [REDACTED] --debug run --tag netgen
Trying https://app.rainforestqa.com/api/1/runs...connected
POST HTTP/1.1
User Agent: rainforest-cli/2.0.4 [rainforest golang lib/2.0.0]
Host: app.rainforestqa.com
```

File: rainforest/rainforest_test.go [hunk 5]
```diff
@@ -137,3 +137,59 @@ func TestDo(t *testing.T) {
...
+			},
+			args:   []string{"rainforest", "--token", "testToken123", "--debug", "run", "--tag", "star"},
+			runID:  564,
+			debug:  true,
+			tag:    "star",
+			token:  "testToken123",
+			method: "GET",
+		},
+		{
+			mappings: map[string]interface{}{
```
Comment (epaulet) [inline rainforest/rainforest_test.go]: Do you still need these mappings?

Comment (sondhayni-rfqa) [inline rainforest/rainforest_test.go]: oh, I suppose not...

**Pull request closed without merging**